protos/agent.proto

```diff
@@ -0,0 +1,45 @@
+/**
+ * proto definitions for auto-generated client, server, and messages to pass using grpc.
+ *
+ * if you make any changes to this file, run
+ *   `rake proto`
+ * to regenerate the ruby code.
+ **/
+
+syntax = "proto3";
+
+package fastlane_c_i;
+
+service Agent {
+  rpc Spawn(Command) returns (stream Log) {}
+}
+
+message Log {
+  // the text contents of the log line.
+  string message = 1;
```

Comment:
Could we mark what is Optional and Required?

`// Optional`
`// Required`

Reply:
Everything is optional in proto3.

Reply:
Yes, but some fields are really required for a service to function appropriately. I'm assuming at some point we'll want to fail the gRPC if the proto in the request is not valid (ex. missing the `bin` field).

Reply:
Hmm, that's a good point. Right now, if you don't supply the value, it will default to an empty-string, and I think that will just return a 126 exit code.